src/sections/layer_and_mask_information_section/mod.rs

```diff
@@ -156,26 +155,24 @@ impl LayerAndMaskInformationSection {
                     _ => unimplemented!("Zip compression currently unsupported"),
                 };
 
-                psd_layer.channels.insert(channel_kind, channel_bytes);
+                psd_layer.channels.insert(*channel_kind, channel_bytes);
             }
 
             match layer_record.divider_type {
                 // open the folder
                 Some(GroupDivider::CloseFolder) | Some(GroupDivider::BoundingSection) => {
-                    nested_level = nested_level + 1;
-                    range_stack.push(idx);
+                    group_start_stack.push(idx);
                 }
 
                 // close the folder
                 Some(GroupDivider::OpenFolder) => {
-                    nested_level = nested_level - 1;
                     let range = Range {
-                        start: range_stack.pop().unwrap(),
+                        start: group_start_stack.pop().unwrap(),
                         end: idx,
                     };
 
                     let group_id = group_id.unwrap();
-                    let parent_group_id = if nested_level > 0 {
+                    let parent_group_id = if group_start_stack.len() > 0 {
                         Some(group_id - 1)
```

Comment:
Could we add a test PSD with the following structure:

```
Can we add a test with the following structure

```
Group
  Group
     Layer
   Group
     Layer
```

And verify that each Group has the correct parent_group_id?

I ask this because right now I think that the logic wouldn't work for this case